src/main.cpp

```diff
@@ -6005,6 +6005,9 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
         {
             boost::this_thread::interruption_point();
 
+            if (!(addr.nServices & NODE_NETWORK))
+                continue;
+
```

Comment:
I agree.  Its either a problem that NODE_NETWORK is set on pruned nodes or that we would not relay them here.  A pruned node is fully capable of engaging in most network behavior.  And in the case of certain VPN pricing, a pruned node might be a better choice -- it may have great network characteristics but its quite expensive to hold but rarely use ancient blockchain data...